### PR TITLE
Added Asciidoc processing class filtering to allow for graceful migrations

### PR DESCRIPTION
**Pull request by johncarl81**
Addresses https://github.com/asciidoctor/asciidoclet/issues/90

### CODE REVIEW

**Comment by johncarl81** [comment]
So, maybe not the best thing, but using `ant-style-path-matcher` required us to drop Java 7 support.

**Comment by johncarl81** [comment]
... thinking about just pulling in the `ant-style-path-matcher` class(es) to downgrade them to Java 7.

**Comment by johncarl81** [comment]
Went ahead and squashed...  slipped my mind that those were defined for javadoc, good catch @gsmet.  Do you have a suggestion for replacement options?

File: README.adoc [hunk 1]
```diff
@@ -239,6 +239,17 @@ Overview documentation can be generated from an Asciidoc file using the standard
 Files matching [x-]`*.adoc`, [x-]`*.ad`, [x-]`*.asciidoc` or [x-]`*.txt` are processed by Asciidoclet.
 Other files are assumed to be HTML and will be processed by the standard doclet.
 
+--include <filter>::
+--exclude <filter>::
+Explicitly include or exclude classes from being processed as Asciidoc comments by ant-style patch matching (see https://github.com/azagniotov/ant-style-path-matcher[ant-style-path-matcher]).
```
Comment (msgilligan) [inline README.adoc]: TYPO: `path` not `patch`

Comment (johncarl81) [reply README.adoc]: Oops, thanks!

**Comment by gsmet** [comment]
As for the options, I think I would have added an asciidoclet prefix to all the asciidoclet specific options. That would make it clearer and avoid potential conflicts.

Something like `--asciidoclet-include`, `--asciidoclet-exclude` and potentially (but unrelated) `--asciidoclet-attribute` and so on (and I would still support the old syntax for compatibility purposes).

The latter point is out of the scope of this PR but I think it would be nice to be consistent.

Currently, the only difference between standard options and specific options is the fact that the standard option has only one dash. It's a bit weird.

**Comment by johncarl81** [comment]
I like being a bit more specific with the options, but I don't think we need to go as far as qualifying them specifically for asciidoclet as there can be only one doclet configured.  How about`--include-asciidoclet-process` and `--exclude-asciidoclet-process`?

**Comment by gsmet** [comment]
@johncarl81 sorry for the late reply. Your proposal looks good to me if you don't want to qualify all the options.

**Comment by johncarl81** [comment]
Ok, made those updates...  final review time @mojavelinux @msgilligan @gsmet ?

**Comment by gsmet** [comment]
@johncarl81 hmmm, totally missed your last message, do you mind updating the PR as it says it's out of date? Once it's done, I'll do some final testing with Hibernate ORM and Quarkus.

Thanks!

**Comment by johncarl81** [comment]
Ok, done @gsmet.  Let me know if you need anything.

**Comment by lptr** [comment]
This would be very useful to us. Any chance to get the PR merged?

**Comment by mojavelinux** [comment]
If there are no objections in the next 24-48 hours, I will merge this PR.

**Comment by johncarl81** [comment]
I like staying somewhat specific here because this filter is for asciidoclet and not the full processing of javadocs.  So, I could see us using `--asciidoclet-include` / `--asciidoclet-exclude` if we want to be a little more concise.  If I get a yay on this, I'll go ahead and make the change.

**Comment by johncarl81** [comment]
And just an unfortunate reminder here.. this only applies to the Java 8 doclet.

**Comment by johncarl81** [comment]
ok, review and merge @mojavelinux ?

**Review by mojavelinux — APPROVED** [review]

**Comment by mojavelinux** [comment]
You can go ahead and merge. I was only jumping in to help...but if you've got it, go for it.

**Comment by mojavelinux** [comment]
:beers:

**Pull request merged**